dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue

```diff
@@ -142,21 +455,143 @@ export default defineComponent({
       try {
         this.postEuTaxonomyDataForNonFinancialsProcessed = false;
         this.messageCount++;
+        let allFilesWasUploadedSuccessful = true;
+
         const euTaxonomyDataForNonFinancialsControllerApi = await new ApiClientProvider(
           assertDefined(this.getKeycloakPromise)()
         ).getEuTaxonomyDataForNonFinancialsControllerApi();
-        this.postEuTaxonomyDataForNonFinancialsResponse =
-          await euTaxonomyDataForNonFinancialsControllerApi.postCompanyAssociatedEuTaxonomyDataForNonFinancials(
-            this.formInputsModel
+
+        const documentUploadControllerControllerApi = await new ApiClientProvider(
+          assertDefined(this.getKeycloakPromise)()
+        ).getDocumentUploadController();
+
+        if (this.filesToUpload.length) {
+          for (let index = 0; index < this.filesToUpload.length; index++) {
+            const uploadFileSuccessful = await documentUploadControllerControllerApi.postDocument(
+              this.filesToUpload[index]
+            );
+            if (!uploadFileSuccessful) {
+              allFilesWasUploadedSuccessful = false;
+              break;
+            } else if (uploadFileSuccessful) {
+              this.updatePropertyFilesUploaded(
+                index,
+                "documentId",
+                uploadFileSuccessful.data.documentId,
+                "filesToUpload"
+              );
+            }
+          }
+        }
+
+        if (allFilesWasUploadedSuccessful) {
+          await this.$nextTick();
+          const formInputsModelToSend = modifyObjectKeys(
+            JSON.parse(JSON.stringify(this.formInputsModel)) as objectType,
+            "send"
           );
+          this.postEuTaxonomyDataForNonFinancialsResponse =
+            await euTaxonomyDataForNonFinancialsControllerApi.postCompanyAssociatedEuTaxonomyDataForNonFinancials(
+              formInputsModelToSend
+            );
+        }
         this.$formkit.reset("createEuTaxonomyForNonFinancialsForm");
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        this.$refs.UploadReports.clearAllNotUploadedFiles();
+        this.fiscalYearEnd = "";
+        this.filesToUpload = [];
+        this.uploadFiles = [];
       } catch (error) {
         this.postEuTaxonomyDataForNonFinancialsResponse = null;
         console.error(error);
       } finally {
         this.postEuTaxonomyDataForNonFinancialsProcessed = true;
       }
     },
```

Comment:
Also in this area please adpot any changes you make in CreateEuTaxonomyForFinancials.vue.